### DISCUSSION TITLE
[Helm]: Should Weblate use StatefulSet instead of Deployment

### DISCUSSION THREAD

**Discussion by mareksuscak** [WeblateOrg/weblate#4806] [Ideas]
I saw the official Helm chart the other day and one thing stood out — it models Weblate as a stateless Deployment rather than StatefulSet which is great for stateful services. As far as I know, Weblate is currently a stateful service and can't be scaled horizontally. We started using Weblate on Kubernetes way before the official Helm chart got released and we first modelled it as Deployment too but the upgrades were somewhat problematic and it kept failing when trying to re-attach persistent disk to the newly spun up container. We used the "Recreate" rollout strategy but it would still fail and then we switched over to StatefulSet and this issue's been gone ever since. 

Anywya, the idea is, should we remodel Weblate as a StatefulService? Is there any specific reason why we're using the Deployment object? I'm assuming that you've already considered it and that there are some reasons that I might have not thought about.

**Comment by nijel** [comment]
What you describe makes sense to me, but my Helm or Kubernetes knowledge is zero :-). The Helm chart was contributed by @tarioch, so maybe he will share his view here.

**Comment by tarioch** [comment]
Hi, I honestly did not think of this too much. 
I changed the stategy to recreate to avoid some of the update issues you mentioned.

Feel free to create a PR to change this, we would probably need to check how this impacts existing deployments so that they are properly upgraded and keep the existing pv.

**Reply by mareksuscak** [reply]
You're absolutely right in that we'd need a solid migration plan. In fact, there's a chance we would need a major version update and have people redeploy weblate which is not great. I'll do the homework and try find a gradual migration path. Thanks for your response, Patrick.
